streaming/java/streaming-api/src/main/java/org/ray/streaming/python/PythonFunction.java

```diff
@@ -88,28 +80,17 @@ public String getFunctionInterface() {
    * @param function serialized python function sent from python driver
    */
   public static PythonFunction fromFunction(byte[] function) {
-    return new PythonFunction(function, null, null, null);
-  }
-
-  /**
-   * Create a {@link PythonFunction} using <code>moduleName</code> and
-   * <code>className</code>.
-   *
-   * @param moduleName python module name
-   * @param className  python class name
-   */
-  public static PythonFunction fromClassName(String moduleName, String className) {
-    return new PythonFunction(null, moduleName, className, null);
+    return new PythonFunction(function, null, null);
   }
 
   /**
    * Create a {@link PythonFunction} using <code>moduleName</code> and
    * <code>functionName</code>.
    *
-   * @param moduleName   python module name
-   * @param functionName python function name
+   * @param moduleName   user function module name
+   * @param functionName function/class name of the user function.
    */
   public static PythonFunction fromFunctionName(String moduleName, String functionName) {
-    return new PythonFunction(null, moduleName, null, functionName);
+    return new PythonFunction(null, moduleName, functionName);
```

Comment:
Again we depend on either function bytes not null or moduleName & functionName not null, it could be replace with `Either` abstraction. Depending on `null` is brittle.

Reply:
I think `Either` is too much for this simple class. It's just a data class that captures some information for a python function. And seems that `Either`  is mainly used for failure/success in `scala` or used in a functional way.  I think we can give more documents and add some assert if needed for now. If user does care about whether this function is constructed from a serialized python function, providing a function like `isSerializedFunction` will be more explicit?

Reply:
Same for `PythonPartition`

Reply:
Fair enough.

The field should be marked as `Nullable` and we have a argument check

```java
checkArgument((function != null) || (moduleName != null && functionName != null));
```

Reply:
Just separate it into two constructor methods